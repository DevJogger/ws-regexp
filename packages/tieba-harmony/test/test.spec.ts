import { escape, unescape } from '../index';

const text = `
废@物裸体果体我无法克制自己对她的不满，于是出声说道
废物裸（河）体果（蟹）体我无法克制自己对她的不满，于是出声说道
廢（·）物裸￥體果 體我無法克制自己對她的不滿，於是出聲說道

废物裸（河）（蟹）体果(和谐)体我无法克制自己对她的不满，于是出声说道
废（·）物裸体果体我无法?克制自己对她的不满，于是出声说道
廢物裸體果體我無法克制自己對她的不滿，於是出聲說道
`;

test(`escape`, () =>
{

	let actual = escape(text);

	expect(actual).toMatchSnapshot();

});

test(`escape`, () =>
{

	let s = escape(text);
	let actual = unescape(s);

	expect(actual).toMatchSnapshot();

});
