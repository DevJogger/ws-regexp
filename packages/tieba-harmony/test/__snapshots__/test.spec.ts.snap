// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`escape 1`] = `
"
废@物裸（河蟹）体果（河蟹）体我无法（河蟹）克制自己对她的不满，于是出声说道
废（河蟹）物裸（河）体果（蟹）体我无法（河蟹）克制自己对她的不满，于是出声说道
廢（·）物裸￥體果 體我無法（河蟹）克制自己對她的不滿，於是出聲說道

废（河蟹）物裸（河）（蟹）体果(和谐)体我无法（河蟹）克制自己对她的不满，于是出声说道
废（·）物裸（河蟹）体果（河蟹）体我无法?克制自己对她的不满，于是出声说道
廢（河蟹）物裸（河蟹）體果（河蟹）體我無法（河蟹）克制自己對她的不滿，於是出聲說道
"
`;

exports[`escape 2`] = `
"
废物裸体裸体我无法克制自己对她的不满，于是出声说道
废物裸体裸体我无法克制自己对她的不满，于是出声说道
廢物裸體裸體我無法克制自己對她的不滿，於是出聲說道

废物裸体裸体我无法克制自己对她的不满，于是出声说道
废物裸体裸体我无法克制自己对她的不满，于是出声说道
廢物裸體裸體我無法克制自己對她的不滿，於是出聲說道
"
`;
